### ISSUE TITLE
kwonly arguments errors

### ISSUE THREAD

**Issue by anntzer** [PyCQA/pylint#1135]
### Steps to reproduce

`foo.py`

```
lambda *, foo: foo

def f(*, foo): return foo
```

then `pylint -E foo.py`
### Current behavior

Outputs

```
No config file found, using default configuration
************* Module foo
E:  1,15: Undefined variable 'foo' (undefined-variable)
E:  3,22: Using variable 'foo' before assignment (used-before-assignment)
```
### Expected behavior

No error should be flagged for either construct.
### pylint --version output

```
No config file found, using default configuration
pylint 1.6.4, 
astroid 1.4.8
Python 3.5.2 (default, Jun 28 2016, 08:46:01) 
[GCC 6.1.1 20160602]
```

**Comment by PCManticore** [comment]
I can reproduce it. Thank you for the report.

**Comment by maliavko** [comment]
Hey guys, are there any plans to fix it? This bug looks very strange, one solution is to disable warning, but there might be real problems in the project with real undefined variables. Looking forward hearing from you.

**Comment by PCManticore** [comment]
Half of this issue was fixed already, didn't manage to find time for it so far.
